packages/relay/src/TransactionManager.ts

```diff
@@ -117,50 +121,76 @@ data         | 0x${transaction.data.toString('hex')}
-        nonce
-      }, this.rawTxOptions)
+      const nonce = await this.pollNonce(txDetails.signer)
+
+      let txToSign: TypedTransaction
+      if (this.transactionType === TransactionType.TYPE_TWO) {
+        // todo add better maxFeePerGas, maxPriorityFeePerGas defaults
+        txToSign = new FeeMarketEIP1559Transaction({
+          to: txDetails.destination,
+          value: txDetails.value,
+          gasLimit: txDetails.gasLimit,
+          maxFeePerGas,
+          maxPriorityFeePerGas: maxPriorityFeePerGas,
+          data: Buffer.from(encodedCall.slice(2), 'hex'),
+          nonce
+        }, this.rawTxOptions)
+      } else {
+        txToSign = new Transaction({
+          to: txDetails.destination,
+          value: txDetails.value,
+          gasLimit: txDetails.gasLimit,
+          gasPrice: maxFeePerGas,
+          data: Buffer.from(encodedCall.slice(2), 'hex'),
+          nonce
+        }, this.rawTxOptions)
+      }
       // TODO omg! do not do this!
-      const keyManager = this.managerKeyManager.isSigner(signer) ? this.managerKeyManager : this.workersKeyManager
-      signedTransaction = keyManager.signTransaction(signer, txToSign)
+      const keyManager = this.managerKeyManager.isSigner(txDetails.signer) ? this.managerKeyManager : this.workersKeyManager
+      signedTransaction = keyManager.signTransaction(txDetails.signer, txToSign)
       const metadata: StoredTransactionMetadata = {
-        from: signer,
+        from: txDetails.signer,
         attempts: 1,
-        serverAction,
-        creationBlockNumber
+        serverAction: txDetails.serverAction,
+        creationBlockNumber: txDetails.creationBlockNumber
       }
       storedTx = createStoredTransaction(signedTransaction.signedEthJsTx, metadata)
-      this.nonces[signer]++
+      this.nonces[txDetails.signer]++
       await this.txStoreManager.putTx(storedTx, false)
-      this.printSendTransactionLog(signedTransaction.signedEthJsTx, signer)
+      this.printSendTransactionLog(storedTx, txDetails.signer)
     } finally {
       releaseMutex()
     }
-    const transactionHash = await this.contractInteractor.broadcastTransaction(signedTransaction.rawTx)
-    if (transactionHash.toLowerCase() !== storedTx.txId.toLowerCase()) {
-      throw new Error(`txhash mismatch: from receipt: ${transactionHash} from txstore:${storedTx.txId}`)
-    }
-    return {
-      transactionHash,
-      signedTx: signedTransaction.rawTx
+    try {
+      const transactionHash = await this.contractInteractor.broadcastTransaction(signedTransaction.rawTx)
+      if (transactionHash.toLowerCase() !== storedTx.txId.toLowerCase()) {
+        throw new Error(`txhash mismatch: from receipt: ${transactionHash} from txstore:${storedTx.txId}`)
+      }
+      return {
+        transactionHash,
+        signedTx: signedTransaction.rawTx
+      }
+    } catch (e) {
+      throw new Error(`Tx broadcast failed: ${(e as Error).message}`)
```

Comment:
thisis an old problem: if we fail to broadcast the transaction - what is next? how we "gracefully " continue ? 
at a minimum, we should flag it as "critical" error, as its not something the server can assume to resolve itself
(there are some known errors it might. currently we ignore it - and sometimes it leaves the server in limbo.)